File: abc/abc163/abc163_b.fs
// https://atcoder.jp/contests/abc163/submissions/12139163

let [| N; _ |] = stdin.ReadLine().Split() |> Array.map int
let A = stdin.ReadLine().Split() |> Array.map int

let days = N - Array.sum A

if days >= 0 then days
else -1
|> stdout.WriteLine

File: abc/abc163/abc163_b.cs
// https://atcoder.jp/contests/abc163/submissions/12141452

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var N = input[0];
        var A = Console.ReadLine().Split().Select(int.Parse).ToArray();

        var days = N - A.Sum();

        Console.WriteLine(days >= 0 ? days : -1);
    }
}
